From 19fc313e37e3e98bb4319f3e475f49a7b58b15c1 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 2 Sep 2015 16:36:38 -0400
Subject: [PATCH 4/7] HACK: ARM: Export cache flush management symbols when
 !MULTI_CACHE

Imagination DDK 1.9 uses dma APIs that are supposed to be private to
dmaengine and not used directly as explained in
https://lkml.org/lkml/2013/1/3/432.

However, this is fixed only in DDK1.12 or later.

For the moment, export the functions as a hack. This patch will be
reverted upon migration to DDK1.14.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
Signed-off-by: Jyri Sarha <jsarha@ti.com>
---
 arch/arm/kernel/setup.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/kernel/setup.c b/arch/arm/kernel/setup.c
index 34e3f3c45634..8213afac8978 100644
--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -1292,3 +1292,12 @@ const struct seq_operations cpuinfo_op = {
 	.stop	= c_stop,
 	.show	= c_show
 };
+
+/* export the cache management functions */
+#ifndef MULTI_CACHE
+
+EXPORT_SYMBOL(__glue(_CACHE, _dma_map_area));
+EXPORT_SYMBOL(__glue(_CACHE, _dma_unmap_area));
+EXPORT_SYMBOL(__glue(_CACHE, _dma_flush_range));
+
+#endif
-- 
2.11.0

